### ISSUE TITLE
Prereq Info incorrect in documentation

### ISSUE THREAD

**Issue by pgierz** [pism/pism#427]
Dear Colleagues,

I'm trying to build PISM1.1 on a CRAY CS400. Unfortunately I'm current getting an error due to an "incorrect" CMake version:
```
CMake Error at CMakeLists.txt:2 (cmake_policy):
  Policy "CMP0078" is not known to this version of CMake.
```

According to the [PISM documentation](http://pism-docs.org/sphinx/installation/prerequisites.html), `CMake` >= 3.1 is needed. [Googling](https://cmake.org/cmake/help/git-stage/policy/CMP0078.html) uncovered that:
>
>"Starting with CMake 3.13,  UseSWIG generates now standard target names. This policy provides compatibility with projects that expect the legacy behavior.” 

We have `CMake --version 3.10` on our machine. I *hope* that this is the only error, but until our system admins update our CMake version, I can't be sure. I've forked and corrected the documentation, once everything builds correctly and I'm sure that's the only thing that is inconsistent , I'll open a merge request.

Cheers,
Paul

**Comment by citibeth** [comment]
Why not try spack for building this software?  With it, you can easily try
a different version of CMake without waiting for your sys admins (or your
sys admins can do these tasks a whole lot easier too).

If we can learn for sure that certain versions of cmake don’t work with
PISM, we can code that into the spack recipe and avoid this same problem in
the future.

Elizabeth

2. Download Spack, play with it, learn how to use it.  It's very easy to
get started, and there is a LOT of help online.
3. Once you're comfortable with Spack, use *spack install pism*.
4. If you encounter any problems, post to the Spack mailing list on Google
Groups.  These people are experts on building / installing software, and
have a wealth of knowledge and experience.

https://spack.io/
https://groups.google.com/forum/#!forum/spack

**Comment by talbrecht** [comment]
I had a similar error with v1.1, which has been solved by the last commit 63dbd9e . Please double check if you have fetched the latest master branch...

**Comment by pgierz** [comment]
Hi Torsten,
I'm specifically using the tagged version 1.1, which points to commit d768cf9b. This one compiles and runs if you use `CMake 3.13.3`. If looks like 63dbd9e checks exactly for this problem. Maybe it would be clever to move the version 1.1 tag?

@citibeth Unfortunately the spack.io page gives me a 503 error, I guess the server is currently offline....

**Comment by aaschwanden** [comment]
@pgierz: PISM compiles for me with make 3.5.2 so as @talbrecht pointed out, commit 
https://github.com/pism/pism/commit/63dbd9e785fc5410659c726f01a2b7136d96f615 should take care of this.

@ckhroulev: can we move the tag?

**Comment by citibeth** [comment]
> @ckhroulev <https://github.com/ckhroulev>: can we move the tag?

Please NEVER move tags (assuming the tag is a releas).  The result is that
you have now made two releases, that are different, but with the same
version number.  That confuses a whole bunch of builder / install systems
(including Spack), which cache downloads and/or assume that checksums for a
given version number remain constant.

Better to add a new tag with a new version number.  If you really don't
want people to use the old tag, you can remove it.

-- Elizabeth

**Comment by pgierz** [comment]
@citibeth Yes, I agree.
"Moving" tags kindof defeats the point of having tags. They should point to unique, machine-identifiable versions (and human-identifiable. Personally, I find 1.1 easier to remember than 63dbd9e), so that automated tools can use them for reference.

Does PISM have some possibility for a continuous integration system? Then it might be possible to avoid this sort of confusion in future. 

For CI, I mean: of course, it shouldn't be a necessity to check every possible PISM build against every possible supercomputer PISM is built on (that's probably not even a realistic possibility). But, it might be an idea to do something like: assuming the documented PISM pre-reqs are installed, PISM should be able to compile and run a minimal experiment (EISMINT II, I guess?). 

I'd be more than happy to help setting something like this up! Have a nice weekend! :-)

**Comment by aaschwanden** [comment]
We've talked about this at UAF a while ago, such capabilities would indeed be useful. Unfortunately, at this time, we don't have sufficient human resources. We'd welcome any help.

**Comment by ckhroulev** [comment]
This issue was fixed in [`v1.1.1`](https://github.com/pism/pism/releases/tag/v1.1.1).
